core/state_transition.go

```diff
@@ -503,19 +504,21 @@ func (st *StateTransition) innerTransitionDb() (*ExecutionResult, error) {
 	// Note for deposit tx there is no ETH refunded for unused gas, but that's taken care of by the fact that gasPrice
 	// is always 0 for deposit tx. So calling refundGas will ensure the gasUsed accounting is correct without actually
 	// changing the sender's balance
+	var gasRefund uint64
 	if !rules.IsLondon {
 		// Before EIP-3529: refunds were capped to gasUsed / 2
-		st.refundGas(params.RefundQuotient)
+		gasRefund = st.refundGas(params.RefundQuotient)
 	} else {
 		// After EIP-3529: refunds are capped to gasUsed / 5
-		st.refundGas(params.RefundQuotientEIP3529)
+		gasRefund = st.refundGas(params.RefundQuotientEIP3529)
 	}
 	if st.msg.IsDepositTx && rules.IsOptimismRegolith {
 		// Skip coinbase payments for deposit tx in Regolith
 		return &ExecutionResult{
-			UsedGas:    st.gasUsed(),
-			Err:        vmerr,
-			ReturnData: ret,
+			UsedGas:     st.gasUsed(),
+			RefundedGas: gasRefund, // TODO(Seb): is this correct for depoit txs?
```

Comment:
Upstream added the refunded gas field to the `ExecutionResult`. Not sure if we want to set this here. IIUC, it might actually be 0 for deposit txs anyways, so inconsequential whether we set it here or not.

Reply:
I'll remove the `TODO` but leave this conversation open to be closed by any reviewer.